core/corehttp/gateway_handler.go

```diff
@@ -174,15 +174,7 @@ func (i *gatewayHandler) getHandler(w http.ResponseWriter, r *http.Request) {
 
 	nd, p, err := i.ResolvePath(ctx, urlPath)
 	if err != nil {
-		if err == routing.ErrNotFound {
-			w.WriteHeader(http.StatusNotFound)
-		} else if err == context.DeadlineExceeded {
-			w.WriteHeader(http.StatusRequestTimeout)
-		} else {
-			w.WriteHeader(http.StatusBadRequest)
-		}
-
-		w.Write([]byte(err.Error()))
+		webError(w, "Path Resolve error", err, http.StatusBadRequest)
```

Comment:
Yep--  what i meant is i missed the `webError` call here -- read it as `w.WriteHeader` still, and the `webError` func was already in the codebase, so the func definition didnt show up in this diff making me realize where the check would do.